lib.js

```diff
@@ -93,3 +93,22 @@ export const generateTestsuiteResult = (suiteResult) => {
 
   return result.join(EOL);
 };
+
+export const generateTestsuiteLogs = (suiteResult) => {
+  let result = [];
+  const testscases = suiteResult.testcase;
+
+  if (testscases) {
+    let errLogs = [...testscases.map(testcase => testcase["system-err"] ? testcase["system-err"].join(EOL).split(EOL).join(EOL + '\t') : "")].filter(a => a != "");
+    if (errLogs.length) {
+      result.push(EOL, "   > Error Log output:", EOL, EOL, '\t', ...errLogs)
+    }
+
+    let outLogs = [...testscases.map(testcase => testcase["system-out"] ? testcase["system-out"].join(EOL).split(EOL).join(EOL + '\t') : "" )].filter(a => a != "")
+    if (outLogs.length) {
+      result.push(EOL, "   > Standard Log output:", EOL, EOL, '\t', ...outLogs)
```

Comment:
Does the EOL need to be pushed here? It could just print the value in the tag, right? The tag output for eg looks like:
```
<system-out>
running 1 test
test test_flaky_mod_4 ... FAILED

failures:

failures:
    test_flaky_mod_4

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 17 filtered out; finished in 0.00s

</system-out>
```
It would usually be already EOL separated. I think it would end up adding unnecessary empty lines between " > Standard Log output:" and the log output.

Reply:
The current code results in formatting that looks like this:

```
> Name:                                    
> Total number of tests:        7          
> Total number of failed tests: 1          
> Total time:                   1531.394346

 FAIL CAPA Standard Suite (1531.394346)
   ✔ [BeforeSuite] (453.040174209)
   ✔ [It] Common tests should be able to connect to MC cluster (0.250346459)
   ✔ [DeferCleanup (Suite)] (764.993963875)

   > Error Log output:

        > Enter [BeforeSuite] TOP-LEVEL - /suite_test.go:35 @ 06/12/23 11:47:37.408
        {"level":"info","ts":"2023-06-12T11:47:37+01:00","msg":"Workload cluster name: t-6z9gz8b"}
        {"level":"info","ts":"2023-06-12T11:47:38+01:00","msg":"Waiting for resource 'org-t-xiy9gae' to be created"}
```

I was struggling to get the right combination of `EOL` and `\t` to keep everything in line 😅  Happy for your help here.

Reply:
@AverageMarcus Would you be able to share a sample junit file, please? It depends on the format of the log output also. For eg, if the log output is formatted like
```
            <system-out>
running 1 test
test test_flaky_mod_4 ... FAILED

failures:

failures:
    test_flaky_mod_4

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 17 filtered out; finished in 0.00s

</system-out>
```
then it won't make sense to reformat and adjust the tabs.

Reply:
Here's a slightly edited / redacted one that I've been using for testing:

```
<?xml version="1.0" encoding="UTF-8"?>
  <testsuites tests="7" disabled="0" errors="0" failures="1" time="1531.394346">
      <testsuite name="Standard Suite" package="/standard" tests="7" disabled="0" skipped="0" errors="0" failures="1" time="1531.394346" timestamp="2023-06-12T11:47:37">
          <properties>
              <property name="SuiteSucceeded" value="false"></property>
              <property name="SuiteHasProgrammaticFocus" value="false"></property>
              <property name="SpecialSuiteFailureReason" value=""></property>
              <property name="SuiteLabels" value="[]"></property>
              <property name="RandomSeed" value="1686566846"></property>
              <property name="RandomizeAllSpecs" value="false"></property>
              <property name="LabelFilter" value=""></property>
              <property name="FocusStrings" value=""></property>
              <property name="SkipStrings" value=""></property>
              <property name="FocusFiles" value=""></property>
              <property name="SkipFiles" value=""></property>
              <property name="FailOnPending" value="false"></property>
              <property name="FailFast" value="false"></property>
              <property name="FlakeAttempts" value="0"></property>
              <property name="DryRun" value="false"></property>
              <property name="ParallelTotal" value="1"></property>
              <property name="OutputInterceptorMode" value=""></property>
          </properties>
          <testcase name="[BeforeSuite]" classname="Standard Suite" status="passed" time="453.040174209">
              <system-err>&gt; Enter [BeforeSuite] TOP-LEVEL - /standard/capa_suite_test.go:35 @ 06/12/23 11:47:37.408&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:47:37+01:00&#34;,&#34;msg&#34;:&#34;Workload cluster name: t-6z9gz8b&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:47:38+01:00&#34;,&#34;msg&#34;:&#34;Waiting for resource &#39;org-t-xiy9gae&#39; to be created&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:47:41+01:00&#34;,&#34;msg&#34;:&#34;Checking for valid Kubeconfig for cluster t-6z9gz8b&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:47:41+01:00&#34;,&#34;msg&#34;:&#34;kubeconfig secret not yet available&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:50:12+01:00&#34;,&#34;msg&#34;:&#34;Checking for valid Kubeconfig for cluster t-6z9gz8b&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:50:12+01:00&#34;,&#34;msg&#34;:&#34;kubeconfig secret not yet available&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:50:22+01:00&#34;,&#34;msg&#34;:&#34;Checking for valid Kubeconfig for cluster t-6z9gz8b&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:50:22+01:00&#34;,&#34;msg&#34;:&#34;kubeconfig secret not yet available&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:50:32+01:00&#34;,&#34;msg&#34;:&#34;Checking for valid Kubeconfig for cluster t-6z9gz8b&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:50:32+01:00&#34;,&#34;msg&#34;:&#34;kubeconfig secret not yet available&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:50:42+01:00&#34;,&#34;msg&#34;:&#34;Checking for valid Kubeconfig for cluster t-6z9gz8b&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:50:42+01:00&#34;,&#34;msg&#34;:&#34;kubeconfig secret not yet available&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:50:52+01:00&#34;,&#34;msg&#34;:&#34;Checking for valid Kubeconfig for cluster t-6z9gz8b&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:53:15+01:00&#34;,&#34;msg&#34;:&#34;Checking for valid Kubeconfig for cluster t-6z9gz8b&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:53:15+01:00&#34;,&#34;msg&#34;:&#34;Got valid kubeconfig!&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:53:15+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:53:15+01:00&#34;,&#34;msg&#34;:&#34;0 nodes ready, expecting 1&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:53:30+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:53:55+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:53:55+01:00&#34;,&#34;msg&#34;:&#34;0 nodes ready, expecting 1&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:54:10+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:54:10+01:00&#34;,&#34;msg&#34;:&#34;0 nodes ready, expecting 1&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:54:40+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:54:40+01:00&#34;,&#34;msg&#34;:&#34;0 nodes ready, expecting 1&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:55:10+01:00&#34;,&#34;msg&#34;:&#34;1 nodes ready, expecting 1&#34;}&#xA;&lt; Exit [BeforeSuite] TOP-LEVEL - /standard/capa_suite_test.go:35 @ 06/12/23 11:55:10.45 (7m33.04s)&#xA;</system-err>
          </testcase>
          <testcase name="[It] Common tests should be able to connect to MC cluster" classname="Standard Suite" status="passed" time="0.250346459">
              <system-err>&gt; Enter [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:55:10.45&#xA;&lt; Exit [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:55:10.45 (0s)&#xA;&gt; Enter [It] should be able to connect to MC cluster - common/common.go:35 @ 06/12/23 11:55:10.45&#xA;&lt; Exit [It] should be able to connect to MC cluster - common/common.go:35 @ 06/12/23 11:55:10.7 (250ms)&#xA;</system-err>
          </testcase>
          <testcase name="[It] Common tests should be able to connect to WC cluster" classname="Standard Suite" status="passed" time="0.013129333">
              <system-err>&gt; Enter [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:55:10.7&#xA;&lt; Exit [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:55:10.701 (0s)&#xA;&gt; Enter [It] should be able to connect to WC cluster - common/common.go:39 @ 06/12/23 11:55:10.701&#xA;&lt; Exit [It] should be able to connect to WC cluster - common/common.go:39 @ 06/12/23 11:55:10.714 (13ms)&#xA;</system-err>
          </testcase>
          <testcase name="[It] Common tests has all of it&#39;s Pods in the Running state" classname="Standard Suite" status="passed" time="242.70246175">
              <system-err>&gt; Enter [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:55:10.714&#xA;&lt; Exit [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:55:10.714 (0s)&#xA;&gt; Enter [It] has all of it&#39;s Pods in the Running state - common/common.go:43 @ 06/12/23 11:55:10.714&#xA;&lt; Exit [It] has all of it&#39;s Pods in the Running state - common/common.go:43 @ 06/12/23 11:59:13.417 (4m2.702s)&#xA;</system-err>
          </testcase>
          <testcase name="[It] Common tests has all the control-plane nodes running" classname="Standard Suite" status="failed" time="0.001326583">
              <failure message="Expected&#xA;    &lt;int&gt;: 1&#xA;to equal&#xA;    &lt;int&gt;: 2" type="failed">[FAILED] Expected&#xA;    &lt;int&gt;: 1&#xA;to equal&#xA;    &lt;int&gt;: 2&#xA;In [It] at: /standard/capa_test.go:19 @ 06/12/23 11:59:13.418&#xA;</failure>
              <system-err>&gt; Enter [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:59:13.418&#xA;&lt; Exit [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:59:13.418 (0s)&#xA;&gt; Enter [It] has all the control-plane nodes running - /standard/capa_test.go:18 @ 06/12/23 11:59:13.418&#xA;[FAILED] Expected&#xA;    &lt;int&gt;: 1&#xA;to equal&#xA;    &lt;int&gt;: 2&#xA;In [It] at: /standard/capa_test.go:19 @ 06/12/23 11:59:13.418&#xA;&lt; Exit [It] has all the control-plane nodes running - /standard/capa_test.go:18 @ 06/12/23 11:59:13.419 (1ms)&#xA;</system-err>
          </testcase>
          <testcase name="[It] Common tests has all the worker nodes running" classname="Standard Suite" status="passed" time="70.391445542">
              <system-err>&gt; Enter [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:59:13.419&#xA;&lt; Exit [BeforeEach] Common tests - common/common.go:28 @ 06/12/23 11:59:13.419 (0s)&#xA;&gt; Enter [It] has all the worker nodes running - /standard/capa_test.go:33 @ 06/12/23 11:59:13.419&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:18+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:18+01:00&#34;,&#34;msg&#34;:&#34;3 nodes ready, expecting between 3 and 10&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:23+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:35+01:00&#34;,&#34;msg&#34;:&#34;3 nodes ready, expecting between 3 and 10&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:35+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:35+01:00&#34;,&#34;msg&#34;:&#34;3 nodes ready, expecting between 3 and 10&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:38+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:38+01:00&#34;,&#34;msg&#34;:&#34;3 nodes ready, expecting between 3 and 10&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:43+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:43+01:00&#34;,&#34;msg&#34;:&#34;3 nodes ready, expecting between 3 and 10&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:48+01:00&#34;,&#34;msg&#34;:&#34;Checking for ready nodes&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T11:59:48+01:00&#34;,&#34;msg&#34;:&#34;3 nodes ready, expecting between 3 and 10&#34;}&#xA;&lt; Exit [It] has all the worker nodes running - /standard/capa_test.go:33 @ 06/12/23 12:00:23.81 (1m10.391s)&#xA;</system-err>
          </testcase>
          <testcase name="[DeferCleanup (Suite)]" classname="Standard Suite" status="passed" time="764.993963875">
              <system-err>&gt; Enter [DeferCleanup (Suite)] TOP-LEVEL - /standard/capa_suite_test.go:76 @ 06/12/23 12:00:23.811&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T12:00:23+01:00&#34;,&#34;msg&#34;:&#34;Checking if resource &#39;t-6z9gz8b&#39; still exists&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T12:00:33+01:00&#34;,&#34;msg&#34;:&#34;Checking if resource &#39;t-6z9gz8b&#39; still exists&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T12:00:43+01:00&#34;,&#34;msg&#34;:&#34;Checking if resource &#39;t-6z9gz8b&#39; still exists&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T12:00:54+01:00&#34;,&#34;msg&#34;:&#34;Checking if resource &#39;t-6z9gz8b&#39; still exists&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T12:01:04+01:00&#34;,&#34;msg&#34;:&#34;Checking if resource &#39;t-6z9gz8b&#39; still exists&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T12:01:14+01:00&#34;,&#34;msg&#34;:&#34;Checking if resource &#39;t-6z9gz8b&#39; still exists&#34;}&#xA;{&#34;level&#34;:&#34;info&#34;,&#34;ts&#34;:&#34;2023-06-12T12:01:24+01:00&#34;,&#34;msg&#34;:&#34;Checking if resource &#39;t-6z9gz8b&#39; still exists&#34;}&#xA;&lt; Exit [DeferCleanup (Suite)] TOP-LEVEL - /standard/capa_suite_test.go:76 @ 06/12/23 12:13:08.806 (12m44.994s)&#xA;</system-err>
          </testcase>
      </testsuite>
  </testsuites>
```

Reply:
Nice, it's now almost good!